js/state/syncUtil.js

```diff
@@ -51,20 +52,27 @@ const pickFields = (object, fields) => {
   }, {})
 }
 
+// Cache of bookmark folder object IDs mapped to folder IDs
+let folderIdMap = new Immutable.Map()
+
 /**
- * Apply a bookmark or historySite SyncRecord to the browser data store.
+ * Converts sync records into a form that can be consumed by AppStore.
  * @param {Object} record
- * @param {Immutable.Map} siteDetail
+ * @param {Immutable.Map} appState
  */
-const applySiteRecord = (record) => {
-  const appActions = require('../actions/appActions')
-  const siteTags = require('../constants/siteTags')
+module.exports.getSiteDataFromRecord = (record, appState) => {
   const objectId = new Immutable.List(record.objectId)
   const category = CATEGORY_MAP[record.objectData].categoryName
-  const existingObject = module.exports.getObjectById(objectId, category)
-  const existingObjectData = existingObject && existingObject[1]
-  let tag
+  let existingObjectData
+
+  if (record.action !== writeActions.CREATE) {
+    // XXX: Is this necessary for CREATEs also?
```

Comment:
omitting the existingObject search for CREATES reduces the sync lag significantly, from a minute or so to a couple seconds.